hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/store/CachedRecordStore.java

```diff
@@ -198,8 +195,15 @@ public void overrideExpiredRecords(QueryResult<R> query) throws IOException {
     if (commitRecords.size() > 0) {
       getDriver().putAll(commitRecords, true, false);
     }
-    if (deleteRecords.size() > 0) {
-      newRecords.removeAll(deleteRecords);
+    if (!toDeleteRecords.isEmpty()) {
+      for (Map.Entry<R, Boolean> entry : getDriver().removeMultiple(toDeleteRecords).entrySet()) {
+        if (entry.getValue()) {
+          deletedRecords.add(entry.getKey());
```

Comment:
Here changing to `newRecords.remove(entry.getKey())`, we can remove  `deletedRecords`.